oxidation/crates/parsec_api_protocol/tests/test_handshake.rs

```diff
@@ -0,0 +1,874 @@
+        "998853e85a67285b0f81c2045be19b7be1c3baad937c5426d68e030ddb46434f2e4d0a560d"
+        "82a6616e73776572c430bbf9777bdc479d6c3d6a77b93aa26370f88ee51f81f3983a074ab3"
+        "2fbfcd33ef6a1b6e3f2e9718e227d0b59e8a749078a474797065ad7369676e65645f616e73"
+        "776572b2636c69656e745f6170695f76657273696f6e920205a96465766963655f6964aa61"
+        "6c6963654064657631a968616e647368616b65a6616e73776572af6f7267616e697a617469"
+        "6f6e5f6964a7436f6f6c4f7267a372766bc420be2976732cec8ca94eedcf0aafd413cd1593"
+        "63e0fadc9e68572c77a1e17d9bbda474797065ad41555448454e54494341544544"
+    );
+
+    let sh = sh.process_answer_req(&answer_req).unwrap();
+
+    assert_eq!(
+        sh.data,
+        Answer::Authenticated {
+            answer: hex!(
+                "aea26853c5d2b206eb3fc532d035164f1b16dea5268b32a3d068fd998853e85a67285b0f81c2045b"
+                "e19b7be1c3baad937c5426d68e030ddb46434f2e4d0a560d82a6616e73776572c430bbf9777bdc47"
+                "9d6c3d6a77b93aa26370f88ee51f81f3983a074ab32fbfcd33ef6a1b6e3f2e9718e227d0b59e8a74"
+                "9078a474797065ad7369676e65645f616e73776572"
+            )
+            .to_vec(),
+            client_api_version: API_V2_VERSION,
+            organization_id: alice.organization_id().to_owned(),
+            device_id: alice.device_id.to_owned(),
+            rvk: Box::new(alice.root_verify_key().to_owned()),
+        }
+    );
+
+    let sh = sh.build_result_req(Some(alice.verify_key())).unwrap();
+
+    assert_eq!(sh.client_api_version, API_V2_VERSION);
+}
+
+#[rstest]
+fn test_good_authenticated_handshake_client(alice: &Device) {
+    // Generated from Python implementation (Parsec v2.6.0+dev)
+    // Content:
+    //   backend_timestamp: ext(1, 1647884434.47775)
+    //   ballpark_client_early_offset: 50.0
+    //   ballpark_client_late_offset: 70.0
+    //   challenge: hex!(
+    //     "58f7ec2bb24b81a57feee1bad250726a2f7588a3cdd0617a206687adf8fb1274f34b0aebf9fd27a5"
+    //     "d29f56dce902ddcd"
+    //   )
+    //   handshake: "challenge"
+    //   supported_api_versions: [[2, 5], [1, 3]
+    let challenge_req = hex!(
+        "86b16261636b656e645f74696d657374616d70d70141d88e2e249e9375bc62616c6c706172"
+        "6b5f636c69656e745f6561726c795f6f6666736574cb4049000000000000bb62616c6c7061"
+        "726b5f636c69656e745f6c6174655f6f6666736574cb4051800000000000a96368616c6c65"
+        "6e6765c43058f7ec2bb24b81a57feee1bad250726a2f7588a3cdd0617a206687adf8fb1274"
+        "f34b0aebf9fd27a5d29f56dce902ddcda968616e647368616b65a96368616c6c656e6765b6"
+        "737570706f727465645f6170695f76657273696f6e7392920205920103"
+    );
+
+    let mut ch = AuthenticatedClientHandshakeStalled::new(
+        alice.organization_id().clone(),
+        alice.device_id.clone(),
+        alice.signing_key.clone(),
+        alice.root_verify_key().clone(),
+    );
+
+    // Fix ballpark failure
+    ch.client_timestamp = DateTime::from_f64_with_us_precision(1647884434.47775);
```

Comment:
This is why we should not have side effect on datetime (cf. https://github.com/Scille/parsec-cloud/pull/2111#discussion_r843721057)

`AuthenticatedClientHandshakeStalled::new` should have a `client_timestamp: Datetime` paremeter

Reply:
Done